hazelcast/src/main/java/com/hazelcast/query/impl/getters/JsonDataGetter.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.query.impl.getters;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonParser;
+import com.hazelcast.nio.serialization.Data;
+
+import java.io.IOException;
+
+import static com.hazelcast.internal.serialization.impl.HeapData.HEAP_DATA_OVERHEAD;
+
+@SuppressWarnings("checkstyle:magicnumber")
+public final class JsonDataGetter extends AbstractJsonGetter {
+
+    public static final JsonDataGetter INSTANCE = new JsonDataGetter();
+
+    private JsonFactory factory = new JsonFactory();
+
+    private JsonDataGetter() {
+        super(null);
+    }
+
+    protected JsonParser createParser(Object obj) throws IOException {
+        Data data = (Data) obj;
+        return factory.createParser(data.toByteArray(), HEAP_DATA_OVERHEAD + 4, data.dataSize() - 4);
```

Comment:
What's this 4 bytes? Can we refer to a static field for it so it's more explanatory and gets updated if it changes?